### ISSUE TITLE
Backup SharePoint Progress Text Truncated

### ISSUE THREAD

**Issue by aldycool** [alcionai/corso#4336]
Hi,

In a standard console, I ran corso Docker container to do SharePoint backup like this:
```sh
docker run --rm --name corso --network corso-network \
-v $CORSO_BASEDIR:/app/corso \
ghcr.io/alcionai/corso:v0.12.0 \
backup create sharepoint \
--site https://XXXXXX.sharepoint.com/sites/XXX \
--aws-access-key none --aws-secret-access-key none \
--azure-tenant-id XXX --azure-client-id XXX --azure-client-secret XXX \
--passphrase XXX \
--json --retain-progress
```
And the console output shows truncated text (the filenames are truncated with ellipsis) like these:
```sh
Logging to file: /root/.cache/corso/logs/2023-09-21T14-06-52Z.log
Connecting to repository:      …
Connecting to M365:         done
Backing Up ∙ https://8vf3bz.sha…
Discovering items to backup:   …
∙ libraries:      0s done
∙ files:      0s done
Discovered 11 items to backup
Backing up data:     45s ∙∙∙
Queuing items   Documents/Gener…
Backing up item A-Tale-of-Two-C…
Backing up item A-Tale-of-Two-C…
Backing up item The-War-of-the-…
Backing up item The-Art-of-War.…
Backing up item The-Island-of-D…
Backing up item The-War-of-the-…
[....SNIP....]
```
How can I have the progress text not truncated? I cannot find the flags / switches anywhere on how to do this. Thanks!

**Comment by aldycool** [comment]
Hi @ryanfkeepers , thank you for your reply.

For anyone that also have the same issue, I think I found the problem. I've mentioned that I used corso Docker container to run the commands, and the outputs are truncated. Now I just tried directly using the binary, and the output turns out to be normal, not truncated.

So I think it got something to do with how Docker constrained the stdout when we do `docker run` command. For now I just stick with the binary executable. Thanks a lot for your help @ryanfkeepers .

**Comment by ryanfkeepers** [comment]
You're very welcome!  And good to know about the docker constraint.  We'll keep that in mind for an improvement, since we definitely want the docker experience to be as good as any other UX.

**Comment by meain** [comment]
**TLDR**: Add `-it` flags to the `docker run` command

@aldycool You should ideally call your docker command with the `-i` and `-t` flag when running interactively. The `-t` flag would attach a tty to the `docker run` command which would help us figure how how wide you terminal is. Without that, corso has no idea as to what the terminal width is and defaults to a smaller max width which causes the output to be truncated. While `-t` would be enough to fix the issue that you are having with truncation, I would also reccomend adding `-i` to make it interactive. This could be useful in cases where you would want to interact with the running command, for example `ctrl+c` to kill corso.

**Comment by aldycool** [comment]
Hi @meain , yes you are correct, adding `-it` flags solved the issue for corso Docker container. Thank you.
